filters/scheduler/fifo_test.go

```diff
@@ -155,14 +424,99 @@ func TestFifo(t *testing.T) {
 			epsilon:      1,
 		},
 		{
-			name: "fifo with reaching max concurrency and queue timeouts",
+			name:       "fifoWithbody simple ok",
+			filterFunc: NewFifoWithBody,
 			args: []interface{}{
 				3,
 				5,
-				"10ms",
+				"1s",
 			},
-			freq:          200,
+			freq:          20,
+			per:           100 * time.Millisecond,
+			backendTime:   1 * time.Millisecond,
+			clientTimeout: time.Second,
+			wantConfig: scheduler.Config{
+				MaxConcurrency: 3,
+				MaxQueueSize:   5,
+				Timeout:        time.Second,
+			},
+			wantParseErr: false,
+			wantOkRate:   1.0,
+			epsilon:      1,
```

Comment:
I think it is better to split filter creation test and behaviour tests. 
I would also propose to define filter via eskip and use eskip.MustParse in the behaviour test.

I started to do a refactoring, you can see the idea here https://github.com/zalando/skipper/compare/filters/scheduler/refactor-fifo-test

Reply:
I've created https://github.com/zalando/skipper/pull/2693